apps/condo/domains/resident/schema/FindUnitsByAddressService.test.js

```diff
@@ -0,0 +1,261 @@
+
+    })
+
+    test('returns empty array if no property map for support', async () => {
+        const payload = { addressKey: propertyWithoutMap.addressKey }
+        const [data] = await findUnitsByAddressByTestClient(supportClient, payload)
+
+        expect(data.units).toHaveLength(0)
+    })
+
+    test('returns empty array if no property map for resident', async () => {
+        const payload = { addressKey: propertyWithoutMap.addressKey }
+        const [data] = await findUnitsByAddressByTestClient(residentClient, payload)
+
+        expect(data.units).toHaveLength(0)
+    })
+
+    test('returns array with units if property map exists for support', async () => {
+        const payload = { addressKey: propertyWithMap.addressKey }
+        const [data] = await findUnitsByAddressByTestClient(supportClient, payload)
+
+        expect(data.units).toHaveLength(propertyWithMapSectionUnits.length + propertyWithMapParkingUnits.length)
+    })
+
+    test('returns array with units if property map exists for resident', async () => {
+        const payload = { addressKey: propertyWithMap.addressKey }
+        const [data] = await findUnitsByAddressByTestClient(residentClient, payload)
+
+        expect(data.units).toHaveLength(propertyWithMapSectionUnits.length + propertyWithMapParkingUnits.length)
+    })
+
+    test('service cannot execute the mutation', async () => {
+        const payload = { addressKey: propertyWithMap.addressKey }
+
+        await expectToThrowAccessDeniedErrorToResult(async () => {
+            await findUnitsByAddressByTestClient(serviceClient, payload)
+        })
+    })
+
+    test('staff cannot execute the mutation', async () => {
+        const payload = { addressKey: propertyWithMap.addressKey }
+
+        await expectToThrowAccessDeniedErrorToResult(async () => {
+            await findUnitsByAddressByTestClient(staffClient, payload)
+        })
+    })
+
+    test('anonymous cannot execute the mutation', async () => {
+        const payload = { addressKey: propertyWithMap.addressKey }
+
+        await expectToThrowAuthenticationErrorToResult(async () => {
+            await findUnitsByAddressByTestClient(anonymousClient, payload)
+        })
+    })
+
+    test('all unit types can be returned', async () => {
+        const [organizationObj] = await registerNewOrganization(supportClient)
+        const [property] = await createTestProperty(supportClient, organizationObj, {
+            isApproved: false,
+            map: {
+                'dv': 1,
+                'type': 'building',
+                'sections': [
+                    {
+                        'id': '1',
+                        'type': 'section',
+                        'index': 1,
+                        'name': 'default section',
+                        'floors': [
+                            {
+                                'id': '1',
+                                'type': 'floor',
+                                'index': 1,
```

Comment:
Do u really need this property. Property utils already contain some maps for test proposes